src/main/java/service/Storage.java

```diff
@@ -0,0 +1,146 @@
+package service;
+
+import duke.Duke;
+import exceptions.DukeException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Storage {
+
+    //private service.TaskList taskList;
+    String filePath;
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    private static boolean isCorrupt(File file) throws FileNotFoundException {
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String task[] = scanner.nextLine().split(" \\| ");
+                //TODO can add layer of check to make sure there is no new line.
+                String type = task[0];
+                String done = task[1];
+                String desc = task[2];
+                String time1 = null;
+                String time2 = null;
+                switch (type) {
+                case "T":
+                    break;
+                case "D":
+                    time1 = task[3];
+                    break;
+                case "E":
+                    time1 = task[3];
+                    time2 = task[4];
+                    break;
+                }
+            }
+            scanner.close();
+            return false;
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            return true;
+        }
+    }
+
+    private static void parseTodoFile(File file, TaskList taskList) throws Exception {
+        Scanner scanner = new Scanner(file);
+        Integer taskCounter = 1;
+        while (scanner.hasNextLine()) {
+            String task[] = scanner.nextLine().split(" \\| ");
+            String type = task[0];
+            String done = task[1];
+            String desc = task[2];
+            String time1 = null;
+            String time2 = null;
+            switch (type) {
+            case "D":
+                time1 = task[3];
+                Duke.processCommand("Deadline " + desc + " /by " + time1, taskList, false);
+                break;
+            case "E":
+                time1 = task[3];
+                time2 = task[4];
+                Duke.processCommand("Event " + desc + " /from " + time1 + " /to " + time2, taskList, false);
+                break;
+            case "T":
+                Duke.processCommand("Todo " + desc, taskList, false);
+                break;
+            }
+
+            if (done.equals("1")) { // done
```

Comment:
Can consider standardising your comment layout, either above code or in the same line as code.